.github/workflows/performance-threshold-update.yml

```
@@ -0,0 +1,215 @@
+  performance-tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 300  # 5 hours for multiple test runs
+    
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      
+      - name: Set up Go
+        uses: actions/setup-go@v4
```

actionlint (1.7.12)

[error] the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)